### PR TITLE
Workflow Client and Its Initial Tests

### PR DESCRIPTION
**Pull request by mindlessroman**
# Description

_Please explain the changes you've made_

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: sdk-workflows/src/main/java/io/dapr/workflows/client/DaprWorkflowClient.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+package io.dapr.workflows.client;
+
+import com.microsoft.durabletask.DurableTaskClient;
+
+public class DaprWorkflowClient {
+
+  private DurableTaskClient innerClient;
+
+  /**
+   * Constructor for DaprWorkflowClient.
```
Comment (mindlessroman) [inline sdk-workflows/src/main/java/io/dapr/workflows/client/DaprWorkflowClient.java:23]: Known TODO: Javadocs have a lot of room for improvement

File: sdk-workflows/src/main/java/io/dapr/workflows/client/DaprWorkflowClient.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+   *
+   * @param innerClient DurableTaskClient instance
+   * @throws IllegalArgumentException catches the nulls
+   */
+  public DaprWorkflowClient(DurableTaskClient innerClient) throws IllegalArgumentException {
+    if (innerClient == null) {
+      throw new IllegalArgumentException("Inner client cannot be null");
```
Comment (mindlessroman) [inline sdk-workflows/src/main/java/io/dapr/workflows/client/DaprWorkflowClient.java:30]: known TODO: Is this the ideal exception to throw and is this messaging.

File: sdk-workflows/src/main/java/io/dapr/workflows/client/DaprWorkflowClient.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+public class DaprWorkflowClient {
+
+  private DurableTaskClient innerClient;
+
+  /**
+   * Constructor for DaprWorkflowClient.
+   *
+   * @param innerClient DurableTaskClient instance
+   * @throws IllegalArgumentException catches the nulls
```
Comment (bderusha) [inline sdk-workflows/src/main/java/io/dapr/workflows/client/DaprWorkflowClient.java]: NIT: should read like a sentence.

```suggestion
   * @throws IllegalArgumentException if innerClient is null
```

File: sdk-workflows/src/main/java/io/dapr/workflows/runtime/DaprWorkflowContextImpl.java [hunk 4]
```diff
@@ -22,8 +22,18 @@ public class DaprWorkflowContextImpl implements WorkflowContext {
 
   private final TaskOrchestrationContext innerContext;
 
-  public DaprWorkflowContextImpl(TaskOrchestrationContext context) {
-    this.innerContext = context;
+  /**
+   * Constructor for DaprWorkflowContextImpl.
+   *
+   * @param context TaskOrchestrationContext
+   * @throws IllegalArgumentException cannot be null
```
Comment (bderusha) [inline sdk-workflows/src/main/java/io/dapr/workflows/runtime/DaprWorkflowContextImpl.java]:
```suggestion
   * @throws IllegalArgumentException if context is null
```

File: sdk-workflows/src/main/java/io/dapr/workflows/runtime/DaprWorkflowContextImpl.java [hunk 5]
```diff
@@ -22,8 +22,18 @@ public class DaprWorkflowContextImpl implements WorkflowContext {
...
+   */
+  public DaprWorkflowContextImpl(TaskOrchestrationContext context) throws IllegalArgumentException {
+    if (context == null) {
+      throw new IllegalArgumentException("Inner context cannot be null"); //TODO: Different exception? Different message?
```
Comment (bderusha) [inline sdk-workflows/src/main/java/io/dapr/workflows/runtime/DaprWorkflowContextImpl.java]:
```suggestion
      throw new IllegalArgumentException("context cannot be null");
```

File: sdk-workflows/src/test/java/io/dapr/workflows/client/DaprWorkflowClientTest.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  @Before
+  public void setUp() throws Exception {
+    DurableTaskClient innerClient = mock(DurableTaskClient.class);
+    when(innerClient.scheduleNewOrchestrationInstance(any())).thenReturn("TestWorkflowInstanceId");
+    client = new DaprWorkflowClient(innerClient);
+  }
+
+  @Test
+  public void scheduleNewWorkflow() {
+    Assert.assertNotNull(client.scheduleNewWorkflow("TestWorkflow"));
```
Comment (bderusha) [inline sdk-workflows/src/test/java/io/dapr/workflows/client/DaprWorkflowClientTest.java:29]: Consider only creating the mock innerClient in the setup and using this test to make a stronger assertion like:

```java
String expectedInstanceId = "TestWorkflowInstanceId";
when(innerClient.scheduleNewOrchestrationInstance(any())).thenReturn(expectedInstanceId);
client = new DaprWorkflowClient(innerClient);
Assert.assertEquals(expectedInstanceId, client.scheduleNewWorkflow("TestWorkflow"));
```

**Review by bderusha — APPROVED** [review]
A few things to fix up, but feel free to merge after that.

File: sdk-workflows/src/test/java/io/dapr/workflows/client/DaprWorkflowClientTest.java (hunk 6 above)
Comment (mindlessroman) [reply sdk-workflows/src/test/java/io/dapr/workflows/client/DaprWorkflowClientTest.java:29]: Valid. Adapting!

**Pull request merged**